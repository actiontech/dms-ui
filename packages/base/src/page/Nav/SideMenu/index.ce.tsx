import {
  DEFAULT_PROJECT_ID,
  DEFAULT_PROJECT_NAME
} from '@actiontech/shared/lib/data/common';
import { Typography } from 'antd';
import ProjectTitle from './ProjectTitle';
import UserMenu from './UserMenu';
import { CEModeProjectWrapperStyleWrapper } from './style';
import { SideMenuStyleWrapper } from '@actiontech/shared/lib/styleWrapper/nav';
import { useCurrentUser } from '@actiontech/shared/lib/global';
import MenuList from './MenuList';
import { FlagFilled } from '@actiontech/icons';

const CESideMenu = () => {
  const { username, theme, updateTheme, language } = useCurrentUser();

  return (
    <SideMenuStyleWrapper className="dms-layout-side">
      <div className="dms-layout-side-start">
        <ProjectTitle />

        <CEModeProjectWrapperStyleWrapper>
          <FlagFilled width={18} height={18} className="project-flag-icon" />
          <Typography.Text className="default-project-name">
            {DEFAULT_PROJECT_NAME}
          </Typography.Text>
        </CEModeProjectWrapperStyleWrapper>

        <MenuList projectID={DEFAULT_PROJECT_ID} />
      </div>

      <UserMenu
        language={language}
        username={username}
        updateTheme={updateTheme}
        theme={theme}
      />
    </SideMenuStyleWrapper>
  );
};

export default CESideMenu;
