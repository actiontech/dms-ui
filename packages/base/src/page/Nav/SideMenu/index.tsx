import { SideMenuStyleWrapper } from './style';
import ProjectSelector from './ProjectSelector';
import { useNavigate } from 'react-router-dom';
import { useMemo } from 'react';
import useRecentlyOpenedProjects from './useRecentlyOpenedProjects';
import { Menu, MenuProps, SelectProps, Typography } from 'antd5';
import {
  useCurrentProject,
  useCurrentUser
} from '@actiontech/shared/lib/global';
import { ProjectSelectorLabelStyleWrapper } from './ProjectSelector/style';
import { CustomSelectProps } from '@actiontech/shared/lib/components/CustomSelect';
import { IconProjectFlag } from '@actiontech/shared/lib/Icon/common';
import {
  DEFAULT_PROJECT_ID,
  DEFAULT_PROJECT_NAME
} from '@actiontech/shared/lib/data/common';
import useSystemConfig from '../../../hooks/useSystemConfig.tsx';
import { SIDE_MENU_DATA_PLACEHOLDER_KEY } from './menus/common';
import { LockOutlined } from '@ant-design/icons';
import UserMenu from './UserMenu';
import ProjectTitle from './ProjectTitle';
import MenuList from './MenuList';

const SideMenu: React.FC = () => {
  const navigate = useNavigate();
  let currentProjectID = '';

  const { username, theme, updateTheme, isAdmin, bindProjects } =
    useCurrentUser();

  const { recentlyProjects, currentProjectID: id } =
    useRecentlyOpenedProjects();

  currentProjectID = id ?? '';

  const projectSelectorOptions = useMemo<SelectProps['options']>(() => {
    return recentlyProjects.map((v) => {
      return {
        value: v.project_id,
        label: (
          <ProjectSelectorLabelStyleWrapper>
            <IconProjectFlag />
            {/* <LockOutlined size={18} /> */}

            <span className="project-selector-label-text">
              {v.project_name}
            </span>
          </ProjectSelectorLabelStyleWrapper>
        ),
        text: v.project_name
      };
    });
  }, [recentlyProjects]);

  const projectSelectorChangeHandle: CustomSelectProps['onChange'] = (id) => {
    navigate(`/sqle/project/${id}/overview`);
  };

  return (
    <SideMenuStyleWrapper className="dms-layout-side">
      <div className="dms-layout-side-start">
        <ProjectTitle />

        <ProjectSelector
          value={currentProjectID}
          onChange={projectSelectorChangeHandle}
          options={projectSelectorOptions}
          bindProjects={bindProjects}
        />

        <MenuList projectID={currentProjectID} isAdmin={isAdmin} />
      </div>

      <UserMenu
        username={username}
        updateTheme={updateTheme}
        isAdmin={isAdmin}
        theme={theme}
      />
    </SideMenuStyleWrapper>
  );
};

export default SideMenu;
