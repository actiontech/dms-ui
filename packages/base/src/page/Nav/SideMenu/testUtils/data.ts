export const mockRecentlyOpenedProjectsData = {
  recentlyProjects: [
    {
      project_id: '1',
      project_name: 'test_project'
    }
  ],
  updateRecentlyProject: jest.fn(),
  currentProjectID: '0'
};
