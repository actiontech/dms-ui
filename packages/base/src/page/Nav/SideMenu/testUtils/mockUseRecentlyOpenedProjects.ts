import * as useRecentlyOpenedProjects from '../useRecentlyOpenedProjects';
import { mockRecentlyOpenedProjectsData } from './data';

export const mockUseRecentlyOpenedProjects = (
  data?: Partial<typeof mockRecentlyOpenedProjectsData>
) => {
  const spy = jest.spyOn(useRecentlyOpenedProjects, 'default');
  spy.mockImplementation(() => ({
    ...mockRecentlyOpenedProjectsData,
    ...data
  }));
  return spy;
};
