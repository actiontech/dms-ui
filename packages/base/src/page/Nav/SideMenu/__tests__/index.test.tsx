import { mockUseCurrentUser } from '@actiontech/shared/lib/testUtil/mockHook/mockUseCurrentUser';
import { mockUseRecentlyOpenedProjects } from '../testUtils/mockUseRecentlyOpenedProjects';
import project from '../../../../testUtils/mockApi/project';
import { superRender } from '../../../../testUtils/customRender';
import SideMenu from '..';
import eventEmitter from '../../../../utils/EventEmitter';
import { act, fireEvent, screen } from '@testing-library/react';
import EmitterKey from '../../../../data/EmitterKey';
import {
  ignoreConsoleErrors,
  UtilsConsoleErrorStringsEnum
} from '@actiontech/shared/lib/testUtil/common';
import { useDispatch } from 'react-redux';
import { getBySelector } from '@actiontech/shared/lib/testUtil/customQuery';
import { useNavigate } from 'react-router-dom';
import { mockProjectList } from '../../../../testUtils/mockApi/project/data';
import { ModalName } from '../../../../data/ModalName';
import { mockUsePermission } from '@actiontech/shared/lib/testUtil/mockHook/mockUsePermission';
import system from '../../../../testUtils/mockApi/system';
import { mockUseRecentlySelectedZone } from '../../../../testUtils/mockHooks/mockUseRecentlySelectedZone';
import gateway from '../../../../testUtils/mockApi/gateway';
import { mockGatewayTipsData } from '../../../../testUtils/mockApi/gateway/data';

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useDispatch: jest.fn()
  };
});

jest.mock('react-router-dom', () => {
  return {
    ...jest.requireActual('react-router-dom'),
    useNavigate: jest.fn()
  };
});

describe('test Base/Nav/SideMenu/index', () => {
  let getProjectsSpy: jest.SpyInstance;
  let subscribeSpy: jest.SpyInstance;
  let getSystemModuleRedDotsSpy: jest.SpyInstance;
  let getGatewayTipsSpy: jest.SpyInstance;

  const dispatchSpy = jest.fn();
  const navigateSpy = jest.fn();
  const unsubscribeSpy = jest.fn();

  const mockBindProjects = mockProjectList.map((v) => ({
    project_id: v.uid ?? '',
    project_name: v.name ?? '',
    archived: !!v.archived,
    is_manager: false
  }));

  ignoreConsoleErrors([UtilsConsoleErrorStringsEnum.INVALID_CUSTOM_ATTRIBUTE]);
  beforeEach(() => {
    subscribeSpy = jest
      .spyOn(eventEmitter, 'subscribe')
      .mockImplementation(() => ({ unsubscribe: unsubscribeSpy }));
    jest.useFakeTimers();
    mockUsePermission(undefined, { useSpyOnMockHooks: true });
    mockUseCurrentUser({
      bindProjects: mockBindProjects
    });
    mockUseRecentlyOpenedProjects({
      currentProjectID: mockBindProjects[0].project_id,
      recentlyProjects: [
        {
          project_id: mockBindProjects[0].project_id,
          project_name: mockBindProjects[0].project_name
        }
      ]
    });

    mockUseRecentlySelectedZone();
    getGatewayTipsSpy = gateway.getGatewayTips();

    getProjectsSpy = project.getProjectList();
    getSystemModuleRedDotsSpy = system.getSystemModuleRedDots();
    (useDispatch as jest.Mock).mockImplementation(() => dispatchSpy);
    (useNavigate as jest.Mock).mockImplementation(() => navigateSpy);
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.clearAllTimers();
    jest.useRealTimers();
  });
  it('mount and unmount component', async () => {
    const { baseElement, unmount } = superRender(<SideMenu />);
    expect(baseElement).toMatchSnapshot();
    expect(getProjectsSpy).toHaveBeenCalledTimes(1);
    expect(getSystemModuleRedDotsSpy).toHaveBeenCalledTimes(1);
    expect(getProjectsSpy).toHaveBeenCalledWith({ page_size: 9999 });
    expect(subscribeSpy).toHaveBeenCalledTimes(2);
    expect(subscribeSpy.mock.calls[0][0]).toBe(
      EmitterKey.Refresh_Availability_Zone_Selector
    );
    expect(subscribeSpy.mock.calls[1][0]).toBe(
      EmitterKey.DMS_Sync_Project_Archived_Status
    );

    await act(async () => jest.advanceTimersByTime(3000));
    expect(dispatchSpy).toHaveBeenNthCalledWith(1, {
      type: 'nav/initModalStatus',
      payload: {
        modalStatus: {
          [ModalName.Company_Notice]: false
        }
      }
    });
    expect(dispatchSpy).toHaveBeenNthCalledWith(2, {
      type: 'user/updateBindProjects',
      payload: {
        bindProjects: mockBindProjects
      }
    });
    expect(baseElement).toMatchSnapshot();

    fireEvent.mouseDown(getBySelector('.ant-select-selector'), baseElement);
    await act(async () => jest.advanceTimersByTime(0));

    expect(baseElement).toMatchSnapshot();

    unmount();
    expect(unsubscribeSpy).toHaveBeenCalledTimes(2);
  });

  it('Click on the menu of the project selector', async () => {
    const { baseElement } = superRender(<SideMenu />);
    await act(async () => jest.advanceTimersByTime(3000));
    fireEvent.mouseDown(getBySelector('.ant-select-selector'), baseElement);
    await act(async () => jest.advanceTimersByTime(300));
    fireEvent.click(screen.getByText(mockBindProjects[1].project_name));
    expect(navigateSpy).toHaveBeenCalledTimes(1);
    expect(navigateSpy).toHaveBeenCalledWith(
      `/sqle/project/${mockBindProjects[1].project_id}/overview`
    );
  });

  it('should dispatch updateAvailabilityZoneTips and call verifyRecentlySelectedZoneRecord func after request gateway tips', async () => {
    const verifyRecentlySelectedZoneRecordSpy = jest.fn();
    mockUseRecentlySelectedZone({
      verifyRecentlySelectedZoneRecord: verifyRecentlySelectedZoneRecordSpy
    });
    superRender(<SideMenu />);
    await act(async () => jest.advanceTimersByTime(3000));
    expect(getGatewayTipsSpy).toHaveBeenCalledTimes(1);
    expect(dispatchSpy).toHaveBeenCalledWith({
      type: 'availabilityZone/updateAvailabilityZoneTips',
      payload: {
        availabilityZoneTips: mockGatewayTipsData
      }
    });
    expect(verifyRecentlySelectedZoneRecordSpy).toHaveBeenCalledTimes(1);
  });
});
