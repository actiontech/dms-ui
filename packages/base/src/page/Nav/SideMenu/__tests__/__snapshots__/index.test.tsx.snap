// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Base/Nav/SideMenu/index mount and unmount component 1`] = `
<body>
  <div>
    <div
      class="dms-layout-side css-13dah2i"
    >
      <div
        class="dms-layout-side-start"
      >
        <div>
          <div
            class="title"
          >
            <span
              class="label label-primary"
            >
              Action
            </span>
            <span
              class="label label-base"
            >
              SQLE
            </span>
          </div>
        </div>
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-txh9fw"
        >
          <div>
            <div
              aria-busy="true"
              aria-live="polite"
              class="ant-spin ant-spin-spinning css-dev-only-do-not-override-txh9fw"
            >
              <span
                class="ant-spin-dot ant-spin-dot-spin"
              >
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
              </span>
            </div>
          </div>
          <div
            class="ant-spin-container ant-spin-blur"
          >
            <div
              class="ant-select ant-select-lg ant-select-borderless basic-select-wrapper custom-select-namespace custom-project-selector css-wk9fc9 css-dev-only-do-not-override-v9sov3 ant-select-single ant-select-show-arrow"
              searchinputref="[object Object]"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                    aria-autocomplete="list"
                    aria-controls="rc_select_TEST_OR_SSR_list"
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-owns="rc_select_TEST_OR_SSR_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="rc_select_TEST_OR_SSR"
                    readonly=""
                    role="combobox"
                    style="opacity: 0;"
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  class="ant-select-selection-item"
                >
                  <span
                    class="custom-select-option-content css-vgjuov"
                  >
                    <span
                      class="custom-select-option-content-prefix"
                    >
                      <svg
                        height="18"
                        viewBox="0 0 18 18"
                        width="18"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M2.25 2.25h7.037a.75.75 0 0 1 .67.415L10.5 3.75H15a.75.75 0 0 1 .75.75v8.25a.75.75 0 0 1-.75.75h-4.786a.75.75 0 0 1-.671-.415L9 12H3.75v4.5h-1.5z"
                          fill="#EBAD1C"
                        />
                      </svg>
                    </span>
                    <span
                      class="custom-select-option-content-label"
                    >
                      test_project_1
                    </span>
                  </span>
                </span>
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <svg
                  class="custom-icon-right-arrow-select-suffix"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m8.781 8-3.3-3.3.943-.943L10.667 8l-4.243 4.243-.943-.943z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <ul
          class="ant-menu ant-menu-root ant-menu-inline ant-menu-light custom-menu css-dev-only-do-not-override-txh9fw"
          data-menu-list="true"
          role="menu"
          tabindex="0"
        />
        <div
          aria-hidden="true"
          style="display: none;"
        />
      </div>
      <div
        class="dms-layout-side-end"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-spin-container"
          >
            <span
              class="ant-avatar ant-avatar-circle action-avatar css-1wq79dc css-dev-only-do-not-override-txh9fw"
            >
              <span
                class="ant-avatar-string"
                style="transform: scale(1) translateX(-50%);"
              >
                A
              </span>
            </span>
          </div>
        </div>
        <div
          class="global-system-icon-wrapper"
        >
          <svg
            class="custom-icon-global-system"
            color=""
            fill="none"
            height="18"
            viewBox="0 0 18 18"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M16.125 4.875 9 .75 1.875 4.875v8.25L9 17.25l7.125-4.125zM9 12.75a3.75 3.75 0 1 0 0-7.5 3.75 3.75 0 0 0 0 7.5"
              fill="#C3C6CD"
              fill-rule="evenodd"
            />
            <path
              d="M9 11.25a2.25 2.25 0 1 0 0-4.5 2.25 2.25 0 0 0 0 4.5"
              fill="#8A8F99"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`test Base/Nav/SideMenu/index mount and unmount component 2`] = `
<body>
  <div>
    <div
      class="dms-layout-side css-13dah2i"
    >
      <div
        class="dms-layout-side-start"
      >
        <div>
          <div
            class="title"
          >
            <span
              class="label label-primary"
            >
              Action
            </span>
            <span
              class="label label-base"
            >
              SQLE
            </span>
          </div>
        </div>
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-select ant-select-lg ant-select-borderless basic-select-wrapper custom-select-namespace custom-project-selector css-wk9fc9 css-dev-only-do-not-override-v9sov3 ant-select-single ant-select-show-arrow"
              searchinputref="[object Object]"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                    aria-autocomplete="list"
                    aria-controls="rc_select_TEST_OR_SSR_list"
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-owns="rc_select_TEST_OR_SSR_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="rc_select_TEST_OR_SSR"
                    readonly=""
                    role="combobox"
                    style="opacity: 0;"
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  class="ant-select-selection-item"
                >
                  <span
                    class="custom-select-option-content css-vgjuov"
                  >
                    <span
                      class="custom-select-option-content-prefix"
                    >
                      <svg
                        height="18"
                        viewBox="64 64 896 896"
                        width="18"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32M332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332zm460 600H232V536h560zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0"
                        />
                      </svg>
                    </span>
                    <span
                      class="custom-select-option-content-label"
                    >
                      test_project_1
                    </span>
                  </span>
                </span>
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <svg
                  class="custom-icon-right-arrow-select-suffix"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m8.781 8-3.3-3.3.943-.943L10.667 8l-4.243 4.243-.943-.943z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <ul
          class="ant-menu ant-menu-root ant-menu-inline ant-menu-light custom-menu css-dev-only-do-not-override-txh9fw"
          data-menu-list="true"
          role="menu"
          tabindex="0"
        />
        <div
          aria-hidden="true"
          style="display: none;"
        />
      </div>
      <div
        class="dms-layout-side-end"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-spin-container"
          >
            <span
              class="ant-avatar ant-avatar-circle action-avatar css-1wq79dc css-dev-only-do-not-override-txh9fw"
            >
              <span
                class="ant-avatar-string"
                style="transform: scale(1) translateX(-50%);"
              >
                A
              </span>
            </span>
          </div>
        </div>
        <div
          class="global-system-icon-wrapper"
        >
          <svg
            class="custom-icon-global-system"
            color=""
            fill="none"
            height="18"
            viewBox="0 0 18 18"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M16.125 4.875 9 .75 1.875 4.875v8.25L9 17.25l7.125-4.125zM9 12.75a3.75 3.75 0 1 0 0-7.5 3.75 3.75 0 0 0 0 7.5"
              fill="#C3C6CD"
              fill-rule="evenodd"
            />
            <path
              d="M9 11.25a2.25 2.25 0 1 0 0-4.5 2.25 2.25 0 0 0 0 4.5"
              fill="#8A8F99"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`test Base/Nav/SideMenu/index mount and unmount component 3`] = `
<body>
  <div>
    <div
      class="dms-layout-side css-13dah2i"
    >
      <div
        class="dms-layout-side-start"
      >
        <div>
          <div
            class="title"
          >
            <span
              class="label label-primary"
            >
              Action
            </span>
            <span
              class="label label-base"
            >
              SQLE
            </span>
          </div>
        </div>
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-select ant-select-lg ant-select-borderless basic-select-wrapper custom-select-namespace custom-project-selector css-wk9fc9 css-dev-only-do-not-override-v9sov3 ant-select-single ant-select-show-arrow ant-select-open"
              searchinputref="[object Object]"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                    aria-autocomplete="list"
                    aria-controls="rc_select_TEST_OR_SSR_list"
                    aria-expanded="true"
                    aria-haspopup="listbox"
                    aria-owns="rc_select_TEST_OR_SSR_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="rc_select_TEST_OR_SSR"
                    readonly=""
                    role="combobox"
                    style="opacity: 0;"
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  class="ant-select-selection-item"
                >
                  <span
                    class="custom-select-option-content css-vgjuov"
                  >
                    <span
                      class="custom-select-option-content-prefix"
                    >
                      <svg
                        height="18"
                        viewBox="64 64 896 896"
                        width="18"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32M332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332zm460 600H232V536h560zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0"
                        />
                      </svg>
                    </span>
                    <span
                      class="custom-select-option-content-label"
                    >
                      test_project_1
                    </span>
                  </span>
                </span>
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <svg
                  class="custom-icon-right-arrow-select-suffix"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m8.781 8-3.3-3.3.943-.943L10.667 8l-4.243 4.243-.943-.943z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <ul
          class="ant-menu ant-menu-root ant-menu-inline ant-menu-light custom-menu css-dev-only-do-not-override-txh9fw"
          data-menu-list="true"
          role="menu"
          tabindex="0"
        />
        <div
          aria-hidden="true"
          style="display: none;"
        />
      </div>
      <div
        class="dms-layout-side-end"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-spin-container"
          >
            <span
              class="ant-avatar ant-avatar-circle action-avatar css-1wq79dc css-dev-only-do-not-override-txh9fw"
            >
              <span
                class="ant-avatar-string"
                style="transform: scale(1) translateX(-50%);"
              >
                A
              </span>
            </span>
          </div>
        </div>
        <div
          class="global-system-icon-wrapper"
        >
          <svg
            class="custom-icon-global-system"
            color=""
            fill="none"
            height="18"
            viewBox="0 0 18 18"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M16.125 4.875 9 .75 1.875 4.875v8.25L9 17.25l7.125-4.125zM9 12.75a3.75 3.75 0 1 0 0-7.5 3.75 3.75 0 0 0 0 7.5"
              fill="#C3C6CD"
              fill-rule="evenodd"
            />
            <path
              d="M9 11.25a2.25 2.25 0 1 0 0-4.5 2.25 2.25 0 0 0 0 4.5"
              fill="#8A8F99"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="ant-select-dropdown ant-slide-up-appear ant-slide-up-appear-prepare ant-slide-up custom-select-popup-wrapper css-dev-only-do-not-override-v9sov3 ant-select-dropdown-placement-bottomLeft"
      style="--arrow-x: 0px; --arrow-y: 0px; left: -1000vw; top: -1000vh; box-sizing: border-box; padding: 0px; width: 240px; min-width: 0;"
    >
      <div>
        <div
          class="css-h34pqn"
        >
          <span
            class="ant-input-affix-wrapper basic-input-wrapper custom-select-popup-search-input css-d0jdfn ant-input-affix-wrapper-borderless css-dev-only-do-not-override-6svbnr"
          >
            <input
              class="ant-input ant-input-borderless css-dev-only-do-not-override-6svbnr"
              placeholder="搜索"
              type="text"
              value=""
            />
            <span
              class="ant-input-suffix"
            >
              <span
                class="ant-input-clear-icon ant-input-clear-icon-hidden"
                role="button"
                tabindex="-1"
              >
                <svg
                  fill="currentColor"
                  height="14"
                  viewBox="0 0 14 14"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m7 6.175 2.887-2.887.825.825L7.825 7l2.887 2.888-.825.824L7 7.825l-2.888 2.887-.824-.824L6.175 7 3.288 4.113l.824-.825z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <div
            class="select-options-group-label"
          >
            最近打开的项目
          </div>
          <div
            class="css-15pnbqu"
          >
            <div
              id="rc_select_TEST_OR_SSR_list"
              role="listbox"
              style="height: 0px; width: 0px; overflow: hidden;"
            >
              <div
                aria-selected="true"
                id="rc_select_TEST_OR_SSR_list_0"
                role="option"
              >
                100121
              </div>
            </div>
            <div
              class="rc-virtual-list"
              style="position: relative;"
            >
              <div
                class="rc-virtual-list-holder"
                style="max-height: 256px; overflow-y: hidden;"
              >
                <div>
                  <div
                    class="rc-virtual-list-holder-inner"
                    style="display: flex; flex-direction: column;"
                  >
                    <div
                      aria-selected="true"
                      class="ant-select-item ant-select-item-option ant-select-item-option-active ant-select-item-option-selected"
                    >
                      <div
                        class="ant-select-item-option-content"
                      >
                        <div
                          class="css-gmbytu"
                        >
                          <svg
                            height="18"
                            viewBox="64 64 896 896"
                            width="18"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32M332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332zm460 600H232V536h560zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0"
                            />
                          </svg>
                          <span
                            class="project-selector-label-text"
                          >
                            test_project_1
                          </span>
                        </div>
                      </div>
                      <span
                        aria-hidden="true"
                        class="ant-select-item-option-state"
                        style="user-select: none;"
                        unselectable="on"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="rc-virtual-list-scrollbar"
                style="width: 8px; top: 0px; bottom: 0px; right: 0px; position: absolute; display: none;"
              >
                <div
                  class="rc-virtual-list-scrollbar-thumb"
                  style="width: 100%; height: 128px; top: 0px; left: 0px; position: absolute; background: rgba(0, 0, 0, 0.5); border-radius: 99px; cursor: pointer; user-select: none;"
                />
              </div>
            </div>
          </div>
          <div
            class="select-options-group-label"
          >
            您所属的项目
          </div>
          <div
            class="css-11lv9el"
          >
            <div
              class="ant-select-item ant-select-item-option"
            >
              <div
                class="ant-select-item-option-content"
              >
                <div
                  class="css-gmbytu"
                >
                  <svg
                    height="18"
                    viewBox="64 64 896 896"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32M332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332zm460 600H232V536h560zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0"
                    />
                  </svg>
                  <span
                    class="project-selector-label-text"
                  >
                    test_project_1
                  </span>
                </div>
              </div>
            </div>
            <div
              class="ant-select-item ant-select-item-option"
            >
              <div
                class="ant-select-item-option-content"
              >
                <div
                  class="css-gmbytu"
                >
                  <svg
                    class="project-flag-icon"
                    color="currentColor"
                    height="18"
                    viewBox="0 0 18 18"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2.25 2.25h7.037a.75.75 0 0 1 .67.415L10.5 3.75H15a.75.75 0 0 1 .75.75v8.25a.75.75 0 0 1-.75.75h-4.786a.75.75 0 0 1-.671-.415L9 12H3.75v4.5h-1.5z"
                      fill="currentColor"
                    />
                  </svg>
                  <span
                    class="project-selector-label-text"
                  >
                    test_project_2
                  </span>
                </div>
              </div>
            </div>
            <div
              class="ant-select-item ant-select-item-option"
            >
              <div
                class="ant-select-item-option-content"
              >
                <div
                  class="css-gmbytu"
                >
                  <svg
                    height="18"
                    viewBox="64 64 896 896"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32M332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332zm460 600H232V536h560zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0"
                    />
                  </svg>
                  <span
                    class="project-selector-label-text"
                  >
                    test_project_3
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="show-more-project-wrapper"
          >
            <a
              href="/project"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-1yfaqef ant-btn-primary basic-button-wrapper css-geipcv"
                type="button"
              >
                <span>
                  查看更多项目
                </span>
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
